Move common http test utilities from testutils to common4j test fixtures

shahzaibj opened the pull request:
Description: Move common http test utilities from testutils to common4j test fixtures.

Why we are moving these classes? We are moving these classes to test fixtures because test fixtures is where they truly belong. The primary purpose of test fixtures is to be able to share test code across modules. 

This is to break https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1770 into multiple smaller PRs.

shahzaibj committed:
Move common http test utilities from testutils to common4j test fixtures

shahzaibj committed:
Merge branch 'dev' into shahzaibj/tests-changes-facilitate-telemetry

iamgusain commented:
Would be nice to update the description with what is being changed here

shahzaibj commented:
> Would be nice to update the description with what is being changed here

I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.

iamgusain commented on common4j/build.gradle:
```diff
@@ -187,10 +213,15 @@ dependencies {
 
-sourceCompatibility = "1.7"
-targetCompatibility = "1.7"
+sourceCompatibility = "1.8"
```
Why is this change needed?

shahzaibj replied:
I think I needed it for some other code I wrote in the bigger PR. So this is a remnant from that but since it belonged in `build.gradle` therefore I included it here as all the `build.gradle` changes were here so it was easier for me to organize the PRs. 

Also I think we've actually already technically been on Java 8 since Melissa's YubyKit changes (YubyKit SDK is on Java 8). We just hadn't explicitly made the change in build.gradle here.

iamgusain commented:
> > Would be nice to update the description with what is being changed here
> 
> I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.

Something like why we are moving the classes will add some context, when we look back at the PR later.

shahzaibj commented:
> > > Would be nice to update the description with what is being changed here
> > 
> > 
> > I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.
> 
> Something like why we are moving the classes will add some context, when we look back at the PR later.

Added this:

Why we are moving these classes? We are moving these classes to test fixtures because test fixtures is where they truly belong. The primary purpose of test fixtures is to be able to share test code across modules.

Test Fixtures is a concept that @p3dr0rv had introduced to the team some time ago, and I think I also covered it again in one of my recent brown-bags as well.

shahzaibj committed:
Merge branch 'dev' into shahzaibj/tests-changes-facilitate-telemetry

shahzaibj committed:
Merge branch 'dev' into shahzaibj/tests-changes-facilitate-telemetry
